corgi/core/migrations/0106_save_container_taxonomy.py

```diff
@@ -0,0 +1,30 @@
+# Generated by Django 3.2.20 on 2023-11-23 04:48
+
+from django.db import migrations
+from corgi.core.models import Component
+from corgi.tasks.common import slow_save_taxonomy as current_slow_save_taxonony
+
+
+def save_container_taxonomy(apps, schema_editor) -> None:
+    for root_container_no_upstreams in (
+        Component.objects.exclude(software_build__isnull=True)
+        .filter(type="OCI", upstreams=None)
```

Comment:
I think we should filter (narrow the search results) before we try to make the DB iterate over and exclude from a (very large) queryset:
```suggestion
        Component.objects.filter(type="OCI", arch="noarch")
```
I'm also not sure filtering based on a many-to-many upstreams field will work efficiently. Maybe it's better to just check `root_container_no_upstreams.upstreams.exists()` in the for-loop below instead.